pkg/controller/ehpa/predict.go

```diff
@@ -126,41 +126,40 @@ func (c *EffectiveHPAController) NewPredictionObject(ehpa *autoscalingapi.Effect
 
 	var predictionMetrics []predictionapi.PredictionMetric
 	for _, metric := range ehpa.Spec.Metrics {
-		// Convert resource metric into prediction metric
-		if metric.Type == autoscalingv2.ResourceMetricSourceType {
-			metricName := utils.GetPredictionMetricName(metric.Resource.Name)
-			if len(metricName) == 0 {
-				continue
+		//get metricIdentifier by metric.Type and metricName
+		var metricIdentifier string
+		switch metric.Type {
+		case autoscalingv2.ResourceMetricSourceType:
+			switch metric.Resource.Name {
+			case "cpu":
+				metricIdentifier = utils.GetMetricIdentifier(metric, v1.ResourceCPU.String())
+			case "memory":
+				metricIdentifier = utils.GetMetricIdentifier(metric, v1.ResourceMemory.String())
 			}
+		case autoscalingv2.ExternalMetricSourceType:
+			metricIdentifier = utils.GetMetricIdentifier(metric, metric.External.Metric.Name)
+		case autoscalingv2.PodsMetricSourceType:
+			metricIdentifier = utils.GetMetricIdentifier(metric, metric.Pods.Metric.Name)
+		}
 
-			predictionMetrics = append(predictionMetrics, predictionapi.PredictionMetric{
-				ResourceIdentifier: metricName,
-				Type:               predictionapi.ResourceQueryMetricType,
-				ResourceQuery:      &metric.Resource.Name,
-				Algorithm: predictionapi.Algorithm{
-					AlgorithmType: ehpa.Spec.Prediction.PredictionAlgorithm.AlgorithmType,
-					DSP:           ehpa.Spec.Prediction.PredictionAlgorithm.DSP,
-					Percentile:    ehpa.Spec.Prediction.PredictionAlgorithm.Percentile,
-				},
-			})
+		if metricIdentifier == "" {
+			continue
 		}
-		// get expressionQuery according to metric.Type
+
+		//get expressionQuery
 		var expressionQuery string
-		var metricName string
-		switch metric.Type {
-		case autoscalingv2.ExternalMetricSourceType:
-			expressionQuery = utils.GetExpressionQuery(metric.External.Metric.Name, ehpa.Annotations)
-			metricName = metric.External.Metric.Name
-		case autoscalingv2.PodsMetricSourceType:
-			expressionQuery = utils.GetExpressionQuery(metric.Pods.Metric.Name, ehpa.Annotations)
-			metricName = metric.Pods.Metric.Name
+		//first get annocation expressionQuery
+		expressionQuery = utils.GetExpressionQueryAnnocation(metricIdentifier, ehpa.Annotations)
```

Comment:
replace GetExpressionQueryAnnocation to GetExpressionQueryAnnotation

Reply:
done